src/processors/other/Tremolo.cpp

```diff
@@ -121,26 +123,46 @@ void Tremolo::processAudio (AudioBuffer<float>& buffer)
     phaseSmooth.setTargetValue (*rateParam * MathConstants<float>::pi / fs);
     waveSmooth.setTargetValue (*waveParam);
 
-    // fill wave buffer (-1, 1)
-    dsp::AudioBlock<float> waveBlock { waveBuffer };
-    dsp::ProcessContextReplacing<float> waveCtx { waveBlock };
-    fillWaveBuffer (waveBlock.getChannelPointer (0), numSamples, phase);
-
-    // shrink range to (0, 1)
-    waveBlock *= 0.5f;
-    waveBlock += 0.5f;
-
-    // apply depth parameter
-    auto depthVal = std::pow (depthParam->load(), 0.33f);
-    depthGainSmooth.setTargetValue (depthVal);
-    waveBlock.multiplyBy (depthGainSmooth);
-    depthAddSmooth.setTargetValue (1.0f - depthVal);
-    addSmoothed (waveBlock, depthAddSmooth);
-    filter.process (waveCtx);
+    if (inputsConnected.contains (1))
```

Comment:
I was using the `getInputModulationPort()` here for a bit, but found using 1 and 0 almost made more sense? Let me know if you think there is a cleaner way to do this.

Reply:
Yeah, I think you're right... the information flow should be like:
- the module informs `getInputModulationPorts()`
- `getInputModulationPorts()` informs the rest of the application

So then the only thing to improve would be getting rid of the magic numbers:
```cpp
class Tremolo
{
    ...
private:
    enum InputPortIndexes
    {
         AudioInput = 0,
         ModulationInput,
    };

    enum OutputPortIndexes
    {
         AudioOutput = 0,
         ModulationOutput,
    };
};

Tremolo::Tremolo()
{
    ...
    routeExternalModulation (ModulationInput, ModulationOutput);
}

void Tremolo::processAudio()
{
    if (inputsConnected.contains (ModulationInput))
        ...
   
    outputBuffers.getReference (ModulationOutput) = ...;
}
```

Actually, we should probably do something similar in the other multi-input/multi-output modules, but that's outside the scope of this PR.

Reply:
Good idea, this makes things more clear readability wise